File: MaybeNewSolutions/题目1047：素数判定.java

import java.util.Scanner;

/** 
 * @ClassName: Main1046 
 * @Description: TODO 
 * @author wangzq
 * @date 2013-8-7 下午04:00:39 
 * 
 * @version 3.0.0 
 */
public class Main {
	public static void main(String[] args) {
		Scanner scanner = new Scanner(System.in);
		while (scanner.hasNext()) {
			int num = scanner.nextInt();
			if (num <=1) {
				System.out.println("no");
			}else {
				boolean result = false;
				for (int i = 2; i < num; i++) {
					if (num%i == 0) {
						result = true;
						break;
					}
				}
				if (result) {
					System.out.println("no");
				}else {
					System.out.println("yes");
				}
			}
			
		}
	}
}	

/**************************************************************
	Problem: 1047
	User: wzqwsrf
	Language: Java
	Result: Accepted
	Time:80 ms
	Memory:15456 kb
****************************************************************/


                        

File: C++/题目1047：素数判定.cpp

// 题目1047：素数判定

/**
 * @author:wangzq
 * @email:wangzhenqing1008@163.com
 * @date:2015-06-30 11:01:52
 * @url:http://ac.jobdu.com/problem.php?pid=1047
 * 解题思路参考csdn:http://blog.csdn.net/u013027996/article/details/36180571
 */

#include <stdio.h>

int main() {
    int num;
    while (scanf("%d", &num) != EOF) {
        if (num <= 1) {
            printf("no\n");
            continue;
        }
        bool flag = false;
        for (int i = 2; i < num; i++) {
            if (num % i == 0) {
                flag = true;
                break;
            }
        }
        printf(flag ? "no\n" : "yes\n");
    }
    return 0;
}
/**************************************************************
Problem: 1047
User: wangzhenqing
Language: C++
Result: Accepted
Time:0 ms
Memory:1020 kb
****************************************************************/


                        

